server/opendp_apps/analysis/tools/dp_histogram_spec.py

```diff
@@ -100,9 +102,19 @@ def get_preprocessor(self):
             # Yes!
             return self.preprocessor
 
+        # TODO: More general type handling
+        if self.var_type == VAR_TYPE_INTEGER:
+            toa = int
+            tia = int
+        else:
+            toa = str
+            tia = str
+
         preprocessor = (
             make_select_column(key=self.col_index, TOA=str) >>
-            make_count_by_categories(categories=self.categories, MO=L1Distance[int], TIA=str)
+            make_cast(TIA=str, TOA=toa) >>
```

Comment:
@ecowan error may be here? e.g. `TIA=str` instead of `make_case(TIA=tia, TOA=toa) >>`?

Reply:
This is here for imputing different types based on the variable info